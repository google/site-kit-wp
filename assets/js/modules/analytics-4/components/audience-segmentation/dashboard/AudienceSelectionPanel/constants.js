/**
 * AudienceSelectionPanel components - constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const AUDIENCE_ADD_GROUP_NOTICE_SLUG =
	'audience-segmentation-add-group-notice';
export const AUDIENCE_SELECTION_PANEL_OPENED_KEY =
	'googlesitekit-audience-selection-panel-opened';
export const AUDIENCE_SELECTION_FORM = 'audience-selection-form';
export const AUDIENCE_SELECTED = 'audience-selected';
export const AUDIENCE_SELECTION_CHANGED = 'audience-selection-changed';
export const AUDIENCE_CREATION_NOTICE_SLUG =
	'audience-segmentation-creation-notice';
export const AUDIENCE_CREATION_SUCCESS_NOTICE_SLUG =
	'audience-segmentation-creation-success-notice';
export const MIN_SELECTED_AUDIENCES_COUNT = 1;
export const MAX_SELECTED_AUDIENCES_COUNT = 3;
export const AUDIENCE_CREATION_EDIT_SCOPE_NOTICE_SLUG =
	'audience-creation-edit-scope-notice';
export const AUDIENCE_CREATION_FORM = 'audience-creation-form';
