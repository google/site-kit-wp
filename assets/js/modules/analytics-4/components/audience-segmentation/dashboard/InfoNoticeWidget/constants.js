/**
 * Constants related to audience segmentation info notices.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * UI key to display audience segmentation info notice.
 *
 * @since 1.135.0
 * @private
 */
export const AUDIENCE_INFO_NOTICE_HIDE_UI =
	'audience-segmentation-info-notice-ui';

/**
 * Slug for audience segmentation info notice.
 *
 * @since 1.130.0
 * @private
 */
export const AUDIENCE_INFO_NOTICE_SLUG = 'audience-segmentation-info-notice';

// Notice slugs: new-visitors, compare-metrics, custom-audiences, purchasers, returning-visitors, compare-new-returning, or compare-cities

/**
 * List of info notices for audiences.
 *
 * @since 1.130.0
 * @private
 */
export const AUDIENCE_INFO_NOTICES = [
	{
		slug: 'new-visitors',
		content: __(
			'The higher the portion of new visitors you have, the more your audience is growing. Looking at what content brings them to your site may give you insights on how to reach even more people.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-metrics',
		content: __(
			'Select up to three visitor groups to display on the dashboard and easily compare metrics between them.',
			'google-site-kit'
		),
	},
	{
		slug: 'custom-audiences',
		content: __(
			'Configure your own custom audiences in Analytics to gain deeper insights into visitor behavior, for example consider creating an “Existing customers” or “Subscribers” segment, depending on what goals you have for your site.',
			'google-site-kit'
		),
	},
	{
		slug: 'purchasers',
		content: __(
			'Select the Purchasers visitor group to gain insights into which visitors bring the most revenue to your site.',
			'google-site-kit'
		),
	},
	{
		slug: 'returning-visitors',
		content: __(
			'The more returning visitors your site has, the stronger and more loyal an audience you’re building. Check which content brings people back to your site - it might help you create a strategy to build a community.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-new-returning',
		content: __(
			'Compare the ratio of “new” to “returning” visitors – this can give you insights on whether you have more people stopping by as a one-off, or more loyal visitors.',
			'google-site-kit'
		),
	},
	{
		slug: 'compare-cities',
		content: __(
			'Check the cities which bring you more new vs more returning visitors – there might be new audiences you could engage with in locations you hadn’t thought about.',
			'google-site-kit'
		),
	},
];
