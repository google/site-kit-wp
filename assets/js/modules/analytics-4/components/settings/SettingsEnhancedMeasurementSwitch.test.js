/**
 * SettingsEnhancedMeasurementSwitch tests.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import {
	act,
	createTestRegistry,
	render,
} from '../../../../../../tests/js/test-utils';
import * as fixtures from '../../datastore/__fixtures__';
import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
import {
	ENHANCED_MEASUREMENT_ENABLED,
	ENHANCED_MEASUREMENT_FORM,
	MODULES_ANALYTICS_4,
	PROPERTY_CREATE,
	WEBDATASTREAM_CREATE,
} from '../../datastore/constants';
import SettingsEnhancedMeasurementSwitch from './SettingsEnhancedMeasurementSwitch';

describe( 'SettingsEnhancedMeasurementSwitch', () => {
	const { webDataStreams, accountSummaries } = fixtures;
	const accounts = accountSummaries.accountSummaries;
	const properties = accounts[ 1 ].propertySummaries;
	const accountID = accounts[ 1 ]._id;
	const propertyID = properties[ 0 ]._id;
	const webDataStreamID = webDataStreams[ 0 ]._id;

	let enhancedMeasurementSettingsMock;
	let registry;

	function setupRegistry() {
		enhancedMeasurementSettingsMock = {
			fileDownloadsEnabled: null,
			name: `properties/${ propertyID }/dataStreams/${ webDataStreamID }/enhancedMeasurementSettings`,
			outboundClicksEnabled: null,
			pageChangesEnabled: null,
			scrollsEnabled: null,
			searchQueryParameter: 'q,s,search,query,keyword',
			siteSearchEnabled: null,
			streamEnabled: false,
			uriQueryParameter: null,
			videoEngagementEnabled: null,
		};

		registry = createTestRegistry();

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetSettings( {
			accountID,
			propertyID,
			webDataStreamID,
		} );

		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetAccountSummaries( {
			accountSummaries: accounts,
			nextPageToken: null,
		} );
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.finishResolution( 'getAccountSummaries', [] );
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.receiveGetWebDataStreams( webDataStreams, {
				propertyID,
			} );
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.finishResolution( 'getWebDataStreams', [ propertyID ] );
	}

	beforeEach( () => {
		setupRegistry();

		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.receiveGetEnhancedMeasurementSettings(
				enhancedMeasurementSettingsMock,
				{ propertyID, webDataStreamID }
			);
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.finishResolution( 'getEnhancedMeasurementSettings', [
				propertyID,
				webDataStreamID,
			] );
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.finishResolution( 'isEnhancedMeasurementStreamAlreadyEnabled', [
				propertyID,
				webDataStreamID,
			] );
	} );

	it( 'should render with the switch defaulting to the on position when enhanced measurement is enabled for the web data stream', async () => {
		await registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setEnhancedMeasurementStreamEnabled(
				propertyID,
				webDataStreamID,
				true
			);

		const { container, getByLabelText } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
			{
				registry,
			}
		);

		expect( container ).toMatchSnapshot();

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		expect( switchControl ).toBeChecked();
	} );

	it( 'should render with the switch defaulting to the off position when enhanced measurement is disabled for the web data stream', async () => {
		await registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setEnhancedMeasurementStreamEnabled(
				propertyID,
				webDataStreamID,
				false
			);

		const { container, getByLabelText } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
			{
				registry,
			}
		);

		expect( container ).toMatchSnapshot();

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		expect( switchControl ).not.toBeChecked();
	} );

	it( 'should not render the switch when enhanced measurement is already enabled for the web data stream', () => {
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.receiveGetEnhancedMeasurementSettings(
				{
					...enhancedMeasurementSettingsMock,
					streamEnabled: true,
				},
				{ propertyID, webDataStreamID }
			);

		const { container, queryByLabelText, getByText } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
			{
				registry,
			}
		);

		expect( container ).toMatchSnapshot();

		expect(
			queryByLabelText( 'Enable enhanced measurement' )
		).not.toBeInTheDocument();

		expect(
			getByText(
				'Enhanced measurement is enabled for this web data stream'
			)
		).toBeInTheDocument();
	} );

	describe.each( [
		[ 'propertyID', PROPERTY_CREATE ],
		[ 'webDataStreamID', WEBDATASTREAM_CREATE ],
	] )( 'when %s is %s', ( settingName, settingCreate ) => {
		beforeEach( () => {
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.setSettings( { [ settingName ]: settingCreate } );
		} );

		it( 'should render correctly, with the switch defaulting to the on position', () => {
			const { container, getByLabelText } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			expect( container ).toMatchSnapshot();

			const switchControl = getByLabelText(
				'Enable enhanced measurement'
			);

			expect( switchControl ).toBeChecked();
		} );

		it( 'should not default the switch to the on position when `isEnhancedMeasurementEnabled` is already `false`', () => {
			registry
				.dispatch( CORE_FORMS )
				.setValues( ENHANCED_MEASUREMENT_FORM, {
					[ ENHANCED_MEASUREMENT_ENABLED ]: false,
				} );

			const { getByLabelText } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			const switchControl = getByLabelText(
				'Enable enhanced measurement'
			);

			expect( switchControl ).not.toBeChecked();
		} );
	} );

	it.each( [
		[
			'enhanced measurement settings is loading',
			() => {
				registry
					.dispatch( MODULES_ANALYTICS_4 )
					.invalidateResolution( 'getEnhancedMeasurementSettings', [
						propertyID,
						webDataStreamID,
					] );
			},
		],
		[
			'property summaries are loading',
			() => {
				registry
					.dispatch( MODULES_ANALYTICS_4 )
					.invalidateResolution( 'getAccountSummaries', [] );
			},
		],
		[
			'web data streams are loading',
			() => {
				registry
					.dispatch( MODULES_ANALYTICS_4 )
					.invalidateResolution( 'getWebDataStreams', [
						propertyID,
					] );
			},
		],
	] )(
		'should render correctly, with the switch in the loading state when %s',
		async ( _, setLoadingState ) => {
			setLoadingState();

			const { container, getByRole, waitForRegistry } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			expect( container ).toMatchSnapshot();

			expect( getByRole( 'progressbar' ) ).toBeInTheDocument();

			await waitForRegistry();
		}
	);

	it( 'should render correctly, with the switch disabled when hasModuleAccess is false', async () => {
		await registry
			.dispatch( MODULES_ANALYTICS_4 )
			.setEnhancedMeasurementStreamEnabled(
				propertyID,
				webDataStreamID,
				false
			);

		const { container, getByLabelText } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess={ false } />,
			{ registry }
		);

		expect( container ).toMatchSnapshot();

		expect(
			getByLabelText( 'Enable enhanced measurement' )
		).toBeDisabled();
	} );

	it( 'should toggle the switch on click', async () => {
		const { getByLabelText, waitForRegistry } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
			{
				registry,
			}
		);

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		expect( switchControl ).not.toBeChecked();

		switchControl.click();

		await waitForRegistry();

		expect( switchControl ).toBeChecked();
	} );

	it( 'should toggle the streamEnabled setting on click', async () => {
		const { getByLabelText, waitForRegistry } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
			{
				registry,
			}
		);

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		expect(
			registry
				.select( MODULES_ANALYTICS_4 )
				.isEnhancedMeasurementStreamEnabled(
					propertyID,
					webDataStreamID
				)
		).toBe( false );

		switchControl.click();

		await waitForRegistry();

		expect(
			registry
				.select( MODULES_ANALYTICS_4 )
				.isEnhancedMeasurementStreamEnabled(
					propertyID,
					webDataStreamID
				)
		).toBe( true );
	} );

	describe.each( [
		[ 'propertyID', PROPERTY_CREATE ],
		[ 'webDataStreamID', WEBDATASTREAM_CREATE ],
	] )( 'when the %s is changed to %s', ( settingName, settingCreate ) => {
		beforeEach( async () => {
			await registry
				.dispatch( MODULES_ANALYTICS_4 )
				.setEnhancedMeasurementStreamEnabled(
					propertyID,
					webDataStreamID,
					true
				);
		} );

		it( 'should revert the switch from off to on', async () => {
			const { getByLabelText, waitForRegistry } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			const switchControl = getByLabelText(
				'Enable enhanced measurement'
			);

			switchControl.click();

			await waitForRegistry();

			expect( switchControl ).not.toBeChecked();

			act( () => {
				registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
					[ settingName ]: settingCreate,
				} );
			} );

			expect( switchControl ).toBeChecked();
		} );

		it( 'should not toggle the switch from on to off', () => {
			const { getByLabelText } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			const switchControl = getByLabelText(
				'Enable enhanced measurement'
			);

			expect( switchControl ).toBeChecked();

			act( () => {
				registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
					[ settingName ]: settingCreate,
				} );
			} );

			expect( switchControl ).toBeChecked();
		} );
	} );

	it( "should set the switch according to the web data stream's streamEnabled value when changing propertyID", async () => {
		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
			propertyID: PROPERTY_CREATE,
		} );
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.receiveGetEnhancedMeasurementSettings(
				{
					...enhancedMeasurementSettingsMock,
					streamEnabled: true,
				},
				{ propertyID, webDataStreamID }
			);

		const { getByLabelText, getByText, waitForRegistry } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
			{
				registry,
			}
		);

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		switchControl.click();

		await waitForRegistry();

		expect( switchControl ).not.toBeChecked();

		act( () => {
			registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
				propertyID,
			} );
		} );

		await waitForRegistry();

		expect( switchControl ).not.toBeInTheDocument();

		expect(
			getByText(
				'Enhanced measurement is enabled for this web data stream'
			)
		).toBeInTheDocument();
	} );

	it( "should set the switch according to the web data stream's streamEnabled value when changing webDataStreamID", async () => {
		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
			webDataStreamID: WEBDATASTREAM_CREATE,
		} );
		registry
			.dispatch( MODULES_ANALYTICS_4 )
			.receiveGetEnhancedMeasurementSettings(
				{
					...enhancedMeasurementSettingsMock,
					streamEnabled: true,
				},
				{ propertyID, webDataStreamID }
			);

		const { getByLabelText, getByText, waitForRegistry } = render(
			<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
			{
				registry,
			}
		);

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		switchControl.click();

		await waitForRegistry();

		expect( switchControl ).not.toBeChecked();

		act( () => {
			registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
				webDataStreamID,
			} );
		} );

		await waitForRegistry();

		expect( switchControl ).not.toBeInTheDocument();

		expect(
			getByText(
				'Enhanced measurement is enabled for this web data stream'
			)
		).toBeInTheDocument();
	} );

	describe( 'synchronization of enhanced measurement settings retrieval with loading states', () => {
		const enhancedMeasurementSettingsEndpoint = new RegExp(
			'^/google-site-kit/v1/modules/analytics-4/data/enhanced-measurement-settings'
		);

		beforeEach( () => {
			setupRegistry();

			fetchMock.getOnce( enhancedMeasurementSettingsEndpoint, {
				status: 200,
				body: enhancedMeasurementSettingsMock,
			} );
		} );

		it( 'should not attempt to retrieve enhanced measurement settings when property summaries are loading', async () => {
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.invalidateResolution( 'getAccountSummaries', [] );

			const { waitForRegistry } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			await waitForRegistry();

			expect( fetchMock ).toHaveFetchedTimes( 0 );
		} );

		it( 'should not attempt to retrieve enhanced measurement settings when web data streams are loading', async () => {
			registry
				.dispatch( MODULES_ANALYTICS_4 )
				.invalidateResolution( 'getWebDataStreams', [ propertyID ] );

			const { waitForRegistry } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			await waitForRegistry();

			expect( fetchMock ).toHaveFetchedTimes( 0 );
		} );

		it( 'should retrieve enhanced measurement settings when neither properties or web data streams are loading', async () => {
			const { waitForRegistry } = render(
				<SettingsEnhancedMeasurementSwitch hasModuleAccess />,
				{
					registry,
				}
			);

			await waitForRegistry();

			expect( fetchMock ).toHaveFetchedTimes( 1 );
			expect( fetchMock ).toHaveFetched(
				enhancedMeasurementSettingsEndpoint
			);
		} );
	} );
} );
