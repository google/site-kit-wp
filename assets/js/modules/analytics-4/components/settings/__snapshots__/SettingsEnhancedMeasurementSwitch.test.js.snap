// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsEnhancedMeasurementSwitch should not render the switch when enhanced measurement is already enabled for the web data stream 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      class="googlesitekit-analytics-enable-enhanced-measurement__already-enabled-label"
    >
      <div
        class="googlesitekit-analytics-enable-enhanced-measurement__already-enabled-tick"
      >
        <svg />
      </div>
      Enhanced measurement is enabled for this web data stream
    </div>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch should render correctly, with the switch disabled when hasModuleAccess is false 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      aria-checked="false"
      class="mdc-switch mdc-switch--disabled"
      role="switch"
      tabindex="0"
    >
      <div
        class="mdc-switch__track"
      >
         
      </div>
      <div
        class="mdc-switch__thumb-underlay"
      >
        <div
          class="mdc-switch__thumb"
        >
          <input
            class="mdc-switch__native-control"
            disabled=""
            id="googlesitekit-switch-7"
            readonly=""
            role="switch"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <label
      class=""
      for="googlesitekit-switch-7"
    >
      Enable enhanced measurement
    </label>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch should render correctly, with the switch in the loading state when enhanced measurement settings is loading 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement googlesitekit-analytics-enable-enhanced-measurement--loading"
  >
    <div
      class="mdc-linear-progress googlesitekit-analytics-enable-enhanced-measurement__progress--settings-edit mdc-linear-progress--indeterminate mdc-linear-progress--small"
      role="progressbar"
    >
      <div
        class="mdc-linear-progress__buffering-dots"
      />
      <div
        class="mdc-linear-progress__buffer"
      />
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
    </div>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch should render correctly, with the switch in the loading state when property summaries are loading 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement googlesitekit-analytics-enable-enhanced-measurement--loading"
  >
    <div
      class="mdc-linear-progress googlesitekit-analytics-enable-enhanced-measurement__progress--settings-edit mdc-linear-progress--indeterminate mdc-linear-progress--small"
      role="progressbar"
    >
      <div
        class="mdc-linear-progress__buffering-dots"
      />
      <div
        class="mdc-linear-progress__buffer"
      />
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
    </div>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch should render correctly, with the switch in the loading state when web data streams are loading 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement googlesitekit-analytics-enable-enhanced-measurement--loading"
  >
    <div
      class="mdc-linear-progress googlesitekit-analytics-enable-enhanced-measurement__progress--settings-edit mdc-linear-progress--indeterminate mdc-linear-progress--small"
      role="progressbar"
    >
      <div
        class="mdc-linear-progress__buffering-dots"
      />
      <div
        class="mdc-linear-progress__buffer"
      />
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
    </div>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch should render with the switch defaulting to the off position when enhanced measurement is disabled for the web data stream 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      aria-checked="false"
      class="mdc-switch"
      role="switch"
      tabindex="0"
    >
      <div
        class="mdc-switch__track"
      >
         
      </div>
      <div
        class="mdc-switch__thumb-underlay"
      >
        <div
          class="mdc-switch__thumb"
        >
          <input
            class="mdc-switch__native-control"
            id="googlesitekit-switch-1"
            readonly=""
            role="switch"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <label
      class=""
      for="googlesitekit-switch-1"
    >
      Enable enhanced measurement
    </label>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch should render with the switch defaulting to the on position when enhanced measurement is enabled for the web data stream 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      aria-checked="true"
      class="mdc-switch mdc-switch--checked"
      role="switch"
      tabindex="0"
    >
      <div
        class="mdc-switch__track"
      >
         
      </div>
      <div
        class="mdc-switch__thumb-underlay"
      >
        <div
          class="mdc-switch__thumb"
        >
          <input
            class="mdc-switch__native-control"
            id="googlesitekit-switch-0"
            readonly=""
            role="switch"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <label
      class=""
      for="googlesitekit-switch-0"
    >
      Enable enhanced measurement
    </label>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch when propertyID is property_create should render correctly, with the switch defaulting to the on position 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      aria-checked="true"
      class="mdc-switch mdc-switch--checked"
      role="switch"
      tabindex="0"
    >
      <div
        class="mdc-switch__track"
      >
         
      </div>
      <div
        class="mdc-switch__thumb-underlay"
      >
        <div
          class="mdc-switch__thumb"
        >
          <input
            class="mdc-switch__native-control"
            id="googlesitekit-switch-2"
            readonly=""
            role="switch"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <label
      class=""
      for="googlesitekit-switch-2"
    >
      Enable enhanced measurement
    </label>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`SettingsEnhancedMeasurementSwitch when webDataStreamID is webdatastream_create should render correctly, with the switch defaulting to the on position 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      aria-checked="true"
      class="mdc-switch mdc-switch--checked"
      role="switch"
      tabindex="0"
    >
      <div
        class="mdc-switch__track"
      >
         
      </div>
      <div
        class="mdc-switch__thumb-underlay"
      >
        <div
          class="mdc-switch__thumb"
        >
          <input
            class="mdc-switch__native-control"
            id="googlesitekit-switch-4"
            readonly=""
            role="switch"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <label
      class=""
      for="googlesitekit-switch-4"
    >
      Enable enhanced measurement
    </label>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;
